jdbc/src/main/java/io/kestra/jdbc/JdbcWorkerTriggerResultQueueService.java

```diff
@@ -0,0 +1,67 @@
+package io.kestra.jdbc;
+
+import io.kestra.core.exceptions.DeserializationException;
+import io.kestra.core.queues.QueueFactoryInterface;
+import io.kestra.core.queues.QueueInterface;
+import io.kestra.core.runners.WorkerTriggerResult;
+import io.kestra.core.utils.Either;
+import io.kestra.jdbc.repository.AbstractJdbcWorkerJobRunningRepository;
+import io.kestra.jdbc.runner.JdbcQueue;
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.inject.qualifiers.Qualifiers;
+import jakarta.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.function.Consumer;
+
+@Singleton
+@Slf4j
+public class JdbcWorkerTriggerResultQueueService {
+    private final JdbcQueue<WorkerTriggerResult> workerTriggerResultQueue;
+    private final AbstractJdbcWorkerJobRunningRepository jdbcWorkerJobRunningRepository;
+    private Runnable queueStop;
+
+    @SuppressWarnings("unchecked")
+    public JdbcWorkerTriggerResultQueueService(ApplicationContext applicationContext) {
+        this.workerTriggerResultQueue = (JdbcQueue<WorkerTriggerResult>) applicationContext.getBean(
+            QueueInterface.class,
+            Qualifiers.byName(QueueFactoryInterface.WORKERTRIGGERRESULT_NAMED));
+        this.jdbcWorkerJobRunningRepository = applicationContext.getBean(AbstractJdbcWorkerJobRunningRepository.class);
```

Comment:
Can't you use standard injection instead of searching manually for the bean?

Reply:
I can't do it for the workerTriggerResultQueue, but did it for the jdbcWorkerJobRunningRepository